sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/LazyAggregateCombineFn.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.extensions.sql.impl;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Iterator;
+import java.util.List;
+import org.apache.beam.sdk.extensions.sql.udf.AggregateFn;
+import org.apache.beam.sdk.transforms.Combine;
+
+/**
+ * {@link org.apache.beam.sdk.transforms.Combine.CombineFn} that wraps an {@link AggregateFn}. The
+ * {@link AggregateFn} is lazily instantiated so it doesn't have to be serialized/deserialized.
```

Comment:
Does this create any additional restrictions over CombineFn? (I didn't see any CombineFn that were taking advantage of the serialization that seems to be occurring...)

Reply:
Discussed this offline. The tl;dr is that we're fine as long as the implementation of `userDefinedAggregateFunctions` is idempotent.